app/PickUpDropOffLocations/Address.php

```diff
@@ -69,9 +69,11 @@ public function setStreet1(?string $street1, bool $combined = false): self
         if ($street1 && $combined) {
             try {
                 $addressBreakdown = AddressSplitter::splitAddress($street1);
-                $street1 = $addressBreakdown['streetName'];
-                $this->setStreetNumber((int) Arr::get($addressBreakdown, 'houseNumberParts.base'));
+                $street1 = Arr::get($addressBreakdown, 'streetName');
+                $number = Arr::get($addressBreakdown, 'houseNumberParts.base');
+                $this->setStreetNumber(empty($number) ? null : (int) $number);
                 $this->setStreetNumberSuffix(Arr::get($addressBreakdown, 'houseNumberParts.extension'));
+                $this->setStreet2(Arr::get($addressBreakdown, 'additionToAddress2'));
```

Comment:
I have a suggestion here actually. Lets first check if street2 is empty and then set the addition. Otherwise lets prepend it instead of overwriting it.

I think the following one-liner can apply for these cases:
```suggestion
                $this->setStreet2(trim(Arr::get($addressBreakdown, 'additionToAddress2') . ' ' . $this->street2));
```

Reply:
I think that would be nice. However, we always call `setStreet2()` after `setStreet1()`. How do we nicely handle:
```
->setStreet1('Street 1 A')
->setStreet2('Building B')
```
I actually regret we made this setter so complex. It could have been a `splitStreet()` function which we would only call in the microservices that actually need this data to be separated. The majority of them implodes everything and we had to manually deactivate the splitting there because of the data going missing or the order of street elements being changed.